pkg/synchronization/controller.go

```diff
@@ -459,41 +465,65 @@ func (c *Controller) getServiceData(user auth.OptionalUser, name voyager.Service
 	return c.ServiceCentral.GetService(context.Background(), user, servicecentral.ServiceName(name))
 }
 
-func (c *Controller) buildNotifications(spec creator_v1.ServiceSpec) (*orch_meta.Notifications, error) {
+func (c *Controller) buildNotifications(spec creator_v1.ServiceSpec) (*orch_meta.Notifications, bool /* retriable */, error) {
+	// Default pagerduty values re-used from Micros config.js
+	notifications := orch_meta.Notifications{
+		Email: spec.EmailAddress(),
+		PagerdutyEndpoint: orch_meta.PagerDuty{
+			Generic:    "5d11612f25b840faaf77422edeff9c76",
+			CloudWatch: "https://events.pagerduty.com/adapter/cloudwatch_sns/v1/124e0f010f214a9b9f30b768e7b18e69",
+		},
+		LowPriorityPagerdutyEndpoint: orch_meta.PagerDuty{
+			Generic:    "5d11612f25b840faaf77422edeff9c76",
+			CloudWatch: "https://events.pagerduty.com/adapter/cloudwatch_sns/v1/124e0f010f214a9b9f30b768e7b18e69",
+		},
+	}
+
 	pdEnvMetadata, ok, err := pagerDutyForEnvType(spec.Metadata.PagerDuty, c.ClusterLocation.EnvType)
 	if err != nil {
-		return nil, errors.Wrap(err, "error building notifications for servicemetadata")
+		return nil, false, errors.Wrap(err, "error building pagerduty notifications for servicemetadata")
 	}
 
 	if ok {
 		mainPD, err := convertPagerDuty(pdEnvMetadata.Main)
 		if err != nil {
-			return nil, errors.Wrap(err, "cannot convert main pagerduty entry")
+			return nil, false, errors.Wrap(err, "cannot convert main pagerduty entry")
 		}
 		lowPriPD, err := convertPagerDuty(pdEnvMetadata.LowPriority)
 		if err != nil {
-			return nil, errors.Wrap(err, "cannot convert low priority pagerduty entry")
+			return nil, false, errors.Wrap(err, "cannot convert low priority pagerduty entry")
 		}
 
-		return &orch_meta.Notifications{
+		notifications = orch_meta.Notifications{
 			Email:                        spec.EmailAddress(),
 			PagerdutyEndpoint:            *mainPD,
 			LowPriorityPagerdutyEndpoint: *lowPriPD,
-		}, nil
+		}
+	}
+	integrations, retriable, err := c.getOpsgenieIntegrations(spec.Metadata.Opsgenie)
+	if err != nil {
+		return nil, retriable, errors.Wrap(err, "failed to create opsgenie notifications")
+	}
+	ogNotifications, err := buildOpsgenieNotifications(integrations, c.ClusterLocation.EnvType)
+	if err == nil {
```

Comment:
This `err == nil` feels weird to me as I usually l like to keep the happy path on the first tab. I might have another go at reworking this function, let me know if you have any ideas.

Reply:
we should probably still log the error?
```
if err != nil {
  // log
} else {
  notifications.OpsgenieIntegrations = ogNotifications
}
```